src/ngraph/runtime/cpu/pass/cpu_layout.cpp

```diff
@@ -1168,6 +1169,70 @@ namespace ngraph
                         set_default_layouts(external_function, node);
                     }
                 }
+
+                template <>
+                void CPULayout::LAYOUT_DECL(ngraph::op::Concat)
+                {
+                    if (runtime::cpu::mkldnn_utils::use_mkldnn_kernel(node.get()))
+                    {
+                        auto concat = static_cast<const ngraph::op::Concat*>(node.get());
+                        auto input0_layout =
+                            runtime::cpu::mkldnn_utils::get_input_mkldnn_format(node.get(), 0);
+                        size_t num_inputs = node->get_input_size();
+                        size_t concat_dim = concat->get_concatenation_axis();
+                        auto result_shape = node->get_output_shape(0);
+                        memory::data_type et = runtime::cpu::mkldnn_utils::get_mkldnn_data_type(
+                            node->get_input_element_type(0));
+                        memory::dims mkldnn_result_shape(result_shape.begin(), result_shape.end());
+                        auto result_desc =
+                            memory::desc(mkldnn_result_shape, et, memory::format::any);
+
+                        std::vector<mkldnn::memory::format> inputs_format;
+                        std::vector<mkldnn::memory::desc> inputs_data_desc;
+                        std::vector<mkldnn::memory::primitive_desc> inputs_pd;
+                        auto& mkldnn_emitter = external_function->get_mkldnn_emitter();
+                        vector<TensorViewWrapper> in;
+                        for (const descriptor::Input& input : node->get_inputs())
+                        {
+                            const descriptor::Output& output = input.get_output();
+                            shared_ptr<descriptor::TensorView> tv = output.get_tensor_view();
+                            in.push_back(TensorViewWrapper(tv, "None"));
+                        }
+                        for (size_t i = 0; i < num_inputs; i++)
+                        {
+                            inputs_format.push_back(
+                                runtime::cpu::mkldnn_utils::get_input_mkldnn_format(concat, i));
+                        }
+                        for (size_t i = 0; i < num_inputs; i++)
+                        {
+                            inputs_data_desc.push_back(
+                                mkldnn_emitter->build_memory_descriptor(in[i], inputs_format[i]));
```

Comment:
Create mkldnn memory descriptors directly instead of going through mkldnn_emitter